test/cluster.sh

```diff
@@ -0,0 +1,91 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests against Knative Serving built from source.
+# It is started by prow for each PR. For convenience, it can also be executed manually.
+
+# If you already have the *_OVERRIDE environment variables set, call
+# this script with the --run-tests arguments and it will start knative in
+# the cluster and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start knative in it, run the tests and delete the
+# cluster.
+
+# Location of istio for the test cluster
+readonly ISTIO_YAML=./third_party/istio-1.0.2/istio-lean.yaml
+
+function create_istio() {
+  echo ">> Bringing up Istio"
+  kubectl apply -f ${ISTIO_YAML}
+}
+
+function create_monitoring() {
+  echo ">> Bringing up Monitoring"
+  kubectl apply -R -f config/monitoring/100-common \
+    -f config/monitoring/150-elasticsearch \
+    -f third_party/config/monitoring/common \
+    -f third_party/config/monitoring/elasticsearch \
+    -f config/monitoring/200-common \
+    -f config/monitoring/200-common/100-istio.yaml
+}
+
+function create_everything() {
+  create_istio
+  echo ">> Bringing up Serving"
+  kubectl apply -f third_party/config/build/release.yaml
+  ko apply -f config/
+  # Due to the lack of Status in Istio, we have to ignore failures in initial requests.
+  #
+  # However, since network configurations may reach different ingress pods at slightly
+  # different time, even ignoring failures for initial requests won't ensure subsequent
+  # requests will succeed all the time.  We are disabling ingress pod autoscaling here
+  # to avoid having too much flakes in the tests.  That would allow us to be stricter
+  # when checking non-probe requests to discover other routing issues.
+  #
+  # We should revisit this when Istio API exposes a Status that we can rely on.
+  # TODO(tcnghia): remove this when https://github.com/istio/istio/issues/882 is fixed.
+  echo ">> Patching Istio"
+  kubectl patch hpa -n istio-system knative-ingressgateway --patch '{"spec": {"maxReplicas": 1}}'
+  create_monitoring
+}
+
+function delete_istio() {
+  echo ">> Bringing down Istio"
+  kubectl delete --ignore-not-found=true -f ${ISTIO_YAML}
+  kubectl delete clusterrolebinding cluster-admin-binding
+}
+
+function delete_monitoring() {
+  echo ">> Bringing down Monitoring"
+  kubectl delete --ignore-not-found=true -f config/monitoring/100-common \
+    -f config/monitoring/150-elasticsearch \
+    -f third_party/config/monitoring/common \
+    -f third_party/config/monitoring/elasticsearch \
+    -f config/monitoring/200-common
+}
+
+function delete_everything() {
+  delete_monitoring
+  echo ">> Bringing down Serving"
+  ko delete --ignore-not-found=true -f config/
+  kubectl delete --ignore-not-found=true -f third_party/config/build/release.yaml
+  delete_istio
+}
+
+function teardown() {
```

Comment:
Better document this function, seens it's `e2e-tests.sh` helper-specific.

Reply:
added

Reply:
I don't see the documentation, but anyway, I realized that this shouldn't go here, as it's script-specific (see my comment below).